ui/src/UtteranceApiService.js

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+/**
+ * Service responsible for saving audio files to an external database. 
+ */
+export class UtteranceApiService {
+
+    /**
+     * Create an UtteranceApiService that can externally save audio files.
+     */
+    constructor() {
+        /**
+         * A Blobstore upload link.
+         * @private
+         */
+        this.blobUrl_; 
+        this.getUploadUrl();
+    }
+
+    /**
+     * Set the blobUrl property to be a Blobstore upload link. 
+     */
+    async getUploadUrl() {
+        const response = await fetch('/blobstore-utterance-upload-link');
```

Comment:
Are we able to reuse the same blob url for multiple uploads?

Reply:
Good catch--I thought I tested multiple concurrent uploads but seemed to have forgot on this round of edits--you cannot use the same URL for multiple uploads, but I have fixed the error and manually tested